src/prover_slow.h

```diff
@@ -82,20 +82,25 @@ std::vector<uint8_t> ProveSlow(integer& D, form& x, uint64_t num_iterations) {
     integer L = root(-D, 4);
     PulmarkReducer reducer;
     form y = form::from_abd(x.a, x.b, D);
-    std::vector<form> intermediates;
-    int k, l;
     int d_bits = D.num_bits();
 
+    int k, l;
     ApproximateParameters(num_iterations, l, k);
     if (k <= 0) k = 1;
     if (l <= 0) l = 1;
-    for (int i = 0; i < num_iterations; i++) {
-        if (i % (k * l) == 0) {
-            intermediates.push_back(y);
+    int kl = k * l;
+
+    uint64_t size_vec = ceil(double(num_iterations) / kl);
```

Comment:
It seems risky to downcast from `uint64_t` to `double`. This should provide an alternative way to calculate the `ceil`:

```uint64_t size_vec = (num_iterations + kl - 1) / kl;```

Reply:
It is indeed!

Reply:
```suggestion
    uint64_t const size_vec = (num_iterations + kl - 1) / kl;
```